ceremonies/src/tests.rs

```diff
@@ -2301,10 +2367,189 @@ fn unregistering_newbie_with_no_participants_works() {
 
 		let alice = account_id(&AccountKeyring::Alice.pair());
 
-		assert_ok!(EncointerCeremonies::unregister_newbie(cid, cindex, &alice));
+		assert!(EncointerCeremonies::remove_participant_from_registry(cid, cindex, &alice).is_err());
 	});
 }
 
+#[test]
+fn upgrade_registration_works() {
+	new_test_ext().execute_with(|| {
+		let cid = register_test_community::<TestRuntime>(None, 0.0, 0.0);
+		let cindex = EncointerScheduler::current_ceremony_index();
+
+		IssuedRewards::<TestRuntime>::insert((cid, cindex - 1), 0, ());
+		let bootstrapper = account_id(&AccountKeyring::Ferdie.pair());
+		EncointerCommunities::insert_bootstrappers(cid, vec![bootstrapper.clone()]);
+		assert!(EncointerBalances::issue(cid, &bootstrapper, NominalIncome::from_num(1)).is_ok());
+
+		let a = AccountKeyring::Alice.pair();
+		let alice = account_id(&a);
+
+		assert_ok!(EncointerCeremonies::register(cid, cindex, &alice, false));
+		assert_eq!(EncointerCeremonies::newbie_count((cid, cindex)), 1);
+		assert_eq!(EncointerCeremonies::newbie_registry((cid, cindex), 1).unwrap(), alice);
+
+		let proof = make_reputable_and_get_proof(&a, cid, cindex - 1);
+		assert_ok!(EncointerCeremonies::upgrade_registration(
+			Origin::signed(alice.clone()),
+			cid,
+			proof
+		));
+
+		assert_eq!(EncointerCeremonies::newbie_count((cid, cindex)), 0);
+		assert_eq!(EncointerCeremonies::reputable_count((cid, cindex)), 1);
+		assert_eq!(EncointerCeremonies::reputable_registry((cid, cindex), 1).unwrap(), alice);
+	});
+}
+
+#[test]
+fn upgrade_fails_if_not_registered_or_not_newbie() {
```

Comment:
We are missing the obvious `upgrade_fails_for_inexistent_community`, `upgrade_fails_in_invalid_phase`.